test/test_sanitizer_log_parser.py

```diff
@@ -0,0 +1,78 @@
+from csv import DictReader
+import os
+from typing import Dict, Optional, Tuple
+
+from colcon_sanitizer_reports.sanitizer_log_parser import SanitizerLogParser
+import pytest
+
+
+class Fixture:
+
+    def __init__(self, resource_name: str) -> None:
+        self.resource_name = resource_name
+        self._parser: Optional[SanitizerLogParser] = None
+
+    @property
+    def resource_path(self) -> str:
+        return os.path.dirname(os.path.abspath(__file__)) + '/resources/' + self.resource_name
+
+    @property
+    def input_log_path(self) -> str:
+        return self.resource_path + '/input.log'
+
+    @property
+    def expected_output_csv_path(self) -> str:
+        return self.resource_path + '/expected_output.csv'
+
+    @property
+    def parser(self) -> SanitizerLogParser:
+        if self._parser is None:
+            parser = SanitizerLogParser()
+            parser.set_package(self.resource_name)
+            with open(self.input_log_path, 'r') as input_log_f_in:
+                for line in input_log_f_in.readlines():
+                    parser.add_line(line)
+
+            self._parser = parser
+            parser.set_package(None)
+
+        return self._parser
+
+    @property
+    def report_csv(self) -> DictReader:
+        return DictReader(self.parser.csv.split('\n'))
+
+    @property
+    def expected_csv(self) -> DictReader:
+        with open(self.expected_output_csv_path, 'r') as expected_output_csv_f_in:
+            return DictReader(expected_output_csv_f_in.read().split('\n'))
+
+
+@pytest.fixture(params=os.listdir(os.path.dirname(os.path.abspath(__file__)) + '/resources'))
+def fixture(request) -> Fixture:
+    return Fixture(request.param)
+
+
+def test_csv_has_field_names(fixture) -> None:
+    assert tuple(fixture.report_csv.fieldnames) == ('package', 'error_name', 'key', 'count')
+
+
+def test_csv_has_output(fixture) -> None:
+    if fixture.resource_name == 'no_errors':
```

Comment:
Be careful not to make too strong assumptions about the code:
https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html

Reply:
I'm a little confused by this comment. This test just shows that log files that have any asan/tsan errors will produce non-zero output in the CSV. The test resource named "no_errors" doesn't include any complete asan/tsan errors so there shouldn't be any CSV output. If I should change something here, could you elaborate a little more?

Reply:
After revising, I think this comment applies pretty well to the test above this one (test_csv_has_field_names). I'm changing that one slightly to be less icky.

Reply:
I'm just generally worried of making assumptions about what files contain.
Having a link between assumptions and file names is a bit of slippery road.
I would do it this way: for edge cases (no error, 1 error, 2 errors), use a small hardcoded string. Large tests can use big external files and in this case we just test invariants (should never throw, if there is string X in log file, we should have this property, etc.)

Goal would be to be able to inject any kind of logs there w/o having to change the test too much.